app/models/reviewable_user.rb

```diff
@@ -36,6 +36,30 @@ def perform_approve_user(performed_by, args)
     create_result(:success, :approved)
   end
 
+  def scrub(performed_by, reason, guardian)
+    # We need to scrub the UserHistory record for when this user was deleted, as well as this reviewable's payload
+    UserHistory
+      .where(action: UserHistory.actions[:delete_user])
+      .where("details LIKE :query", query: "%\nusername: #{payload["username"]}\n%")
+      .update_all(
+        details:
+          I18n.t(
+            "user.destroy_reasons.reviewable_details_scrubbed",
+            staff: performed_by.username,
+            reason: reason,
+          ),
+      )
```

Comment:
I guess we cannot use `target_user_id` as user does not exist anymore. But maybe we can make scope even more detailed to avoid the situation that the wrong history item is picked. Edge case when user with same username was rejected year ago.

For example, we can look for UserHistory which was created within 10 minutes of time when reviewable was performed?

Reply:
Yah, I think that's a reasonable approach to limit collateral damage here.